tests/command.init.test.js

```diff
@@ -0,0 +1,283 @@
+const { Buffer } = require('buffer')
+
+const test = require('ava')
+const execa = require('execa')
+const toml = require('toml')
+
+const { readFileAsync } = require('../src/lib/fs')
+
+const cliPath = require('./utils/cli-path')
+const { withMockApi } = require('./utils/mock-api')
+const { withSiteBuilder } = require('./utils/site-builder')
+
+const handleQuestions = (process, questions) => {
+  const remainingQuestions = [...questions]
+  let buffer = ''
+  process.stdout.on('data', (data) => {
+    buffer += data
+    const index = remainingQuestions.findIndex(({ question }) => buffer.includes(question))
+    if (index >= 0) {
+      buffer = ''
+      process.stdin.write(Buffer.from(remainingQuestions[index].answer))
+      remainingQuestions.splice(index, 1)
+    }
+  })
+}
+
+const CONFIRM = '\n'
+const DOWN = '\u001B[B'
+const answerWithValue = (value) => `${value}${CONFIRM}`
+
+const assertSiteInit = async (
+  t,
+  builder,
+  requests,
+  { command = 'custom-build-command', functions = 'custom-functions', publish = 'custom-publish', plugins = [] } = {},
+) => {
+  // assert netlify.toml was created with user inputs
+  const netlifyToml = toml.parse(await readFileAsync(`${builder.directory}/netlify.toml`, 'utf8'))
+  t.deepEqual(netlifyToml, {
+    build: { command, functions, publish },
+    ...(plugins.length === 0 ? {} : { plugins }),
+  })
+
+  // assert updateSite was called with user inputs
+  const siteUpdateRequests = requests.filter(({ path }) => path === '/api/v1/sites/site_id').map(({ body }) => body)
+  t.deepEqual(siteUpdateRequests, [
+    {
+      plugins,
+      repo: {
+        allowed_branches: ['master'],
+        cmd: command,
+        dir: publish,
+        provider: 'manual',
+        repo_branch: 'master',
+        repo_path: 'git@github.com:owner/repo.git',
+      },
+    },
+    {
+      build_settings: {
+        functions_dir: functions,
+      },
+    },
+  ])
+}
+
+test('netlify init existing site', async (t) => {
+  const initQuestions = [
```

Comment:
Figured it's time for us to be able to test the CLI input prompts. If you think of a better way to do this please let me know.

Reply:
This is great! 🎉